core/api/src/services/mongoose/quiz.ts

```diff
@@ -0,0 +1,38 @@
+import { Quiz } from "./schema"
+
+import { QuizAlreadyPresentError, UnknownRepositoryError } from "@/domain/errors"
+
+interface ExtendedError extends Error {
+  code?: number
+}
+
+export const QuizRepository = (accountId: AccountId) => {
```

Comment:
please move accountId to methods, also this requires an interface

Reply:
so that `add` would take `{quizId, accountId}` input instead?

Reply:
yes